test/directed-requester.js

```diff
@@ -0,0 +1,91 @@
+import test from 'ava';
+import r from 'randomstring';
+import LogSuppress from '../lib/log-suppress';
+
+LogSuppress.init(console);
+
+const environment = r.generate();
+const { DirectedRequester, Responder } = require('../')({ environment });
+
+test.cb('Supports directed requests using callbacks', (t) => {
+    const key = r.generate();
+    const namespace = r.generate();
+
+    const requester = new DirectedRequester(
+        { name: `PBR ${t.title}: directed requester`, key, namespace });
+    const responders = [];
+
+    Array.from({ length: 10 }).forEach(() => {
+        responders.push(new Responder({ name: `PBR ${t.title}: directed responder`, key, namespace }));
+    });
+
+    let randomResponder = responders[Math.floor(Math.random() * responders.length)];
+
+    randomResponder.sock.on('bind', () => {
+        const args = [1, 2, 6];
+        requester.send({ responderId: randomResponder.discovery.me.id, type: 'test', args }, (res) => {
```

Comment:
passing in a responderId is great, but it's pretty low level. Would using a human-settable property also work? Would this be more generalizable and be made more user-friendly if we added a new property to the advertisement message (alongside the `name` — surprisingly this is a _pass-on_ object so you can extend it with arbitrary information) like another identifier or a key that is known to both responders and requesters? I'm thinking about the general usability, maybe this would end up being sort of a channelling component where you'd load balance between two responders if they shared that key. This would open up more use-cases for this component.

What do you think?

Reply:
This is a great idea, i incorporated it into the PR